src/bramble/client/ProjectLimiter.js

```diff
@@ -0,0 +1,69 @@
+define(function() {
+    "use strict";
+    var ProjectLimiter = {};
+
+    var _maxProjectState = {
+            fileSize: 0, // max size of any file in bytes. NOT USED now.
+            fileCount: 0, // total number of files you can have in a project
+            projectSize: 0// total number of bytes for all files in this project
+    };
+
+    var _currentProjectState = {
+        numOfFiles: 0 ,
+        projectSize: 0
+    };
+
+    var _fs;
+
+    ProjectLimiter.initMaxLimits =  function (projectState,fs){
+        _fs = fs;
+        _maxProjectState = projectState;
+        console.log(projectState);
+    };
+
+    ProjectLimiter.newFile = function(){
+            _currentProjectState.numOfFiles +=1;
+    };
+
+    //accepts path be checked,the oldSize of a file, and an optional callback.
+    //path has to be pointing to an existing file or to a file that has been just deleted. If not triggers incorrect behaviour.
+    ProjectLimiter.checkLimits = function (pathname,oldSize,callback){
```

Comment:
Because we'll do a mix of walking the root dir at startup, and also wrapping all modifying fs calls after that, we can make methods that return data synchronous: we'll just be looking at values in our cache, or calculating things based on it.

Probably we want to split this method into a bunch of smaller, more specific stats.  Of the top of my head:

```js
ProjectStats.getTotalSize = function(){...} // returns bytes of files in the root
...
ProjectStats.getLargestFileSize = function(type) {
  if(type) {
    // get largest file size for the given type (e.g., image, html, ...).
    // We could do this by extension (.html) or maybe by name (image)
  } else {
    // get larges file size for any type of file (e.g., ignore type)
  }
};
...
ProjectStats.getFileCount = function(type) {
  if(type) {
    // get file count for the given type (e.g., image, html, ...).
    // We could do this by extension (.html) or maybe by name (image)
  } else {
    // get file count for any type of file (e.g., ignore type)
  }
};
...
ProjectStats.hasIndexFile = function() {
  // returns true/false depending on whether root/index.html exists
  // we would love to know this when we publish, so we can warn users.
};
...
ProjectStats.usesCSS = function() {
  // returns true/false depending on whether the project has CSS files
};
...
ProjectStats.usesJS = function() {
  // returns true/false depending on whether the project has JS files
};
...
etc
```